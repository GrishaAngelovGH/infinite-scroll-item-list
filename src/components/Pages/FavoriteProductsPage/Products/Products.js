import { useSelector } from 'react-redux'
import { selectFavoriteItemRows } from 'selectors/itemRowsSelector'

import { Col, Empty, Row } from 'antd'

import ProductItem from 'components/ProductItem'

const Products = () => {
  const itemsRows = useSelector(selectFavoriteItemRows)

  if (!itemsRows.length) {
    return (
      <Empty
        imageStyle={{
          marginTop: '10%',
        }}
        description={
          <h1 style={{ fontSize: 40 }}>
            No items found
          </h1>
        }
      />
    )
  }

  return (
    <Row className='product-catalog'>
      <Col span={24}>
        {
          itemsRows.map((row, i) => (
            <Row key={i}>
              {
                row.map(v => {
                  return (
                    <ProductItem
                      key={v.id}
                      id={v.id}
                      title={v.title}
                      image={v.image}
                      number={v.number}
                      price={v.price}
                      isFavorite={v.isFavorite}
                    />
                  )
                })
              }
            </Row>
          ))
        }
      </Col>
    </Row>
  )
}

export default Products