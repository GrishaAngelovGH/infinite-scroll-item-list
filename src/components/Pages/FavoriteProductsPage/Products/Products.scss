.filter-btn {
  margin: 10px;
  height: 40px;
  border-radius: 8px;

  svg {
    font-size: 30px;
  }
}
