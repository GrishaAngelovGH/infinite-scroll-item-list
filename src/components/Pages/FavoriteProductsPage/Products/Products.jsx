import { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { clearFilters } from 'slices/filtersSlice'

import { selectFavoriteItemRowsByFilter } from 'selectors/itemRowsSelector'

import { ControlFilled, CloseCircleFilled } from '@ant-design/icons'
import { Button, Col, Empty, Row, Tooltip } from 'antd'

import FilterDrawer from 'components/Pages/FavoriteProductsPage/Products/FilterDrawer'
import ProductItem from 'components/ProductItem'

import './Products.scss'

const Products = () => {
  const itemRows = useSelector(selectFavoriteItemRowsByFilter)
  const [open, setOpen] = useState(false)
  const dispatch = useDispatch()

  const showDrawer = () => {
    setOpen(true)
  }

  const onClose = () => {
    setOpen(false)
  }

  const handleClearFilters = () => {
    dispatch(clearFilters())
  }

  return (
    <Row className='product-catalog'>
      <Tooltip title='Filters'>
        <Button
          className='filter-btn'
          type='primary'
          onClick={showDrawer}
        >
          <ControlFilled />
        </Button>
      </Tooltip>

      <Tooltip title='Clear All Filters'>
        <Button
          className='filter-btn'
          type='primary'
          danger
          onClick={handleClearFilters}
        >
          <CloseCircleFilled />
        </Button>
      </Tooltip>

      <FilterDrawer open={open} onClose={onClose} />

      <Col span={24}>
        {
          !itemRows.length && (
            <Empty
              description={
                <h1 style={{ fontSize: 40 }}>
                  No items found
                </h1>
              }
            />
          )
        }
        {
          itemRows.map((row, i) => (
            <Row key={i}>
              {
                row.map(v => {
                  return (
                    <ProductItem
                      key={v.id}
                      id={v.id}
                      title={v.title}
                      image={v.image}
                      number={v.number}
                      price={v.price}
                      isFavorite={v.isFavorite}
                    />
                  )
                })
              }
            </Row>
          ))
        }
      </Col>
    </Row>
  )
}

export default Products