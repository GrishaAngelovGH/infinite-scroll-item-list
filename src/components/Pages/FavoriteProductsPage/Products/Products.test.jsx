import { render, waitFor } from '@testing-library/react'

import { Provider } from 'react-redux'

import {
  Route,
  BrowserRouter as Router,
  Routes
} from 'react-router-dom'

import Products from './Products'

test('should render Products component', async () => {
  await waitFor(() => {
    const view = render(
      <Provider store={globalThis.mockStore}>
        <Router>
          <Routes>
            <Route
              path='/'
              element={<Products />}
            />
          </Routes>
        </Router>
      </Provider>
    )

    expect(view).toMatchSnapshot()
  })
})
