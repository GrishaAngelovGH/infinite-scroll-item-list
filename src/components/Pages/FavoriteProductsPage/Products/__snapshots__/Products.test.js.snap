// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Products component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-row product-catalog css-dev-only-do-not-override-yp8pcc"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-yp8pcc ant-btn-primary filter-btn"
          type="button"
        >
          <span
            aria-label="control"
            class="anticon anticon-control"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="control"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM404 683v77c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-77c-41.7-13.6-72-52.8-72-99s30.3-85.5 72-99V264c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v221c41.7 13.6 72 52.8 72 99s-30.3 85.5-72 99zm279.6-143.9c.2 0 .3-.1.4-.1v221c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V539c.2 0 .3.1.4.1-42-13.4-72.4-52.7-72.4-99.1 0-46.4 30.4-85.7 72.4-99.1-.2 0-.3.1-.4.1v-77c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v77c-.2 0-.3-.1-.4-.1 42 13.4 72.4 52.7 72.4 99.1 0 46.4-30.4 85.7-72.4 99.1zM616 440a36 36 0 1072 0 36 36 0 10-72 0zM403.4 566.5l-1.5-2.4c0-.1-.1-.1-.1-.2l-.9-1.2c-.1-.1-.2-.2-.2-.3-1-1.3-2-2.5-3.2-3.6l-.2-.2c-.4-.4-.8-.8-1.2-1.1-.8-.8-1.7-1.5-2.6-2.1h-.1l-1.2-.9c-.1-.1-.3-.2-.4-.3-1.2-.8-2.5-1.6-3.9-2.2-.2-.1-.5-.2-.7-.4-.4-.2-.7-.3-1.1-.5-.3-.1-.7-.3-1-.4-.5-.2-1-.4-1.5-.5-.4-.1-.9-.3-1.3-.4l-.9-.3-1.4-.3c-.2-.1-.5-.1-.7-.2-.7-.1-1.4-.3-2.1-.4-.2 0-.4 0-.6-.1-.6-.1-1.1-.1-1.7-.2-.2 0-.4 0-.7-.1-.8 0-1.5-.1-2.3-.1s-1.5 0-2.3.1c-.2 0-.4 0-.7.1-.6 0-1.2.1-1.7.2-.2 0-.4 0-.6.1-.7.1-1.4.2-2.1.4-.2.1-.5.1-.7.2l-1.4.3-.9.3c-.4.1-.9.3-1.3.4-.5.2-1 .4-1.5.5-.3.1-.7.3-1 .4-.4.2-.7.3-1.1.5-.2.1-.5.2-.7.4-1.3.7-2.6 1.4-3.9 2.2-.1.1-.3.2-.4.3l-1.2.9h-.1c-.9.7-1.8 1.4-2.6 2.1-.4.4-.8.7-1.2 1.1l-.2.2a54.8 54.8 0 00-3.2 3.6c-.1.1-.2.2-.2.3l-.9 1.2c0 .1-.1.1-.1.2l-1.5 2.4c-.1.2-.2.3-.3.5-2.7 5.1-4.3 10.9-4.3 17s1.6 12 4.3 17c.1.2.2.3.3.5l1.5 2.4c0 .1.1.1.1.2l.9 1.2c.1.1.2.2.2.3 1 1.3 2 2.5 3.2 3.6l.2.2c.4.4.8.8 1.2 1.1.8.8 1.7 1.5 2.6 2.1h.1l1.2.9c.1.1.3.2.4.3 1.2.8 2.5 1.6 3.9 2.2.2.1.5.2.7.4.4.2.7.3 1.1.5.3.1.7.3 1 .4.5.2 1 .4 1.5.5.4.1.9.3 1.3.4l.9.3 1.4.3c.2.1.5.1.7.2.7.1 1.4.3 2.1.4.2 0 .4 0 .6.1.6.1 1.1.1 1.7.2.2 0 .4 0 .7.1.8 0 1.5.1 2.3.1s1.5 0 2.3-.1c.2 0 .4 0 .7-.1.6 0 1.2-.1 1.7-.2.2 0 .4 0 .6-.1.7-.1 1.4-.2 2.1-.4.2-.1.5-.1.7-.2l1.4-.3.9-.3c.4-.1.9-.3 1.3-.4.5-.2 1-.4 1.5-.5.3-.1.7-.3 1-.4.4-.2.7-.3 1.1-.5.2-.1.5-.2.7-.4 1.3-.7 2.6-1.4 3.9-2.2.1-.1.3-.2.4-.3l1.2-.9h.1c.9-.7 1.8-1.4 2.6-2.1.4-.4.8-.7 1.2-1.1l.2-.2c1.1-1.1 2.2-2.4 3.2-3.6.1-.1.2-.2.2-.3l.9-1.2c0-.1.1-.1.1-.2l1.5-2.4c.1-.2.2-.3.3-.5 2.7-5.1 4.3-10.9 4.3-17s-1.6-12-4.3-17c-.1-.2-.2-.4-.3-.5z"
              />
            </svg>
          </span>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-yp8pcc ant-btn-primary ant-btn-dangerous filter-btn"
          type="button"
        >
          <span
            aria-label="close-circle"
            class="anticon anticon-close-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
              />
            </svg>
          </span>
        </button>
        <div
          class="ant-col ant-col-24 css-dev-only-do-not-override-yp8pcc"
        >
          <div
            class="ant-row css-dev-only-do-not-override-yp8pcc"
          >
            <div
              class="ant-col ant-col-sm-24 ant-col-md-8 css-dev-only-do-not-override-yp8pcc"
            >
              <div
                class="ant-card ant-card-bordered product-item css-dev-only-do-not-override-yp8pcc"
              >
                <div
                  class="ant-card-cover"
                >
                  <a
                    href="/123"
                  >
                    <img
                      src="bluetooth-speaker.jpg"
                    />
                  </a>
                </div>
                <div
                  class="ant-card-body"
                >
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-title"
                      >
                        <div
                          class="product-title"
                        >
                          Bluetooth Speaker
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-title"
                      >
                        $123.45
                      </div>
                    </div>
                  </div>
                </div>
                <ul
                  class="ant-card-actions"
                >
                  <li
                    style="width: 50%;"
                  >
                    <span>
                      <span
                        class="product-number"
                      >
                        #
                        1
                      </span>
                    </span>
                  </li>
                  <li
                    style="width: 50%;"
                  >
                    <span>
                      <span
                        aria-label="star"
                        class="anticon anticon-star"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="star"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                          />
                        </svg>
                      </span>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="ant-col ant-col-sm-24 ant-col-md-8 css-dev-only-do-not-override-yp8pcc"
            >
              <div
                class="ant-card ant-card-bordered product-item css-dev-only-do-not-override-yp8pcc"
              >
                <div
                  class="ant-card-cover"
                >
                  <a
                    href="/234"
                  >
                    <img
                      src="gamepad.jpg"
                    />
                  </a>
                </div>
                <div
                  class="ant-card-body"
                >
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-title"
                      >
                        <div
                          class="product-title"
                        >
                          Gamepad
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-title"
                      >
                        $234.56
                      </div>
                    </div>
                  </div>
                </div>
                <ul
                  class="ant-card-actions"
                >
                  <li
                    style="width: 50%;"
                  >
                    <span>
                      <span
                        class="product-number"
                      >
                        #
                        2
                      </span>
                    </span>
                  </li>
                  <li
                    style="width: 50%;"
                  >
                    <span>
                      <span
                        aria-label="star"
                        class="anticon anticon-star"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="star"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                          />
                        </svg>
                      </span>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-row product-catalog css-dev-only-do-not-override-yp8pcc"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-yp8pcc ant-btn-primary filter-btn"
        type="button"
      >
        <span
          aria-label="control"
          class="anticon anticon-control"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="control"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM404 683v77c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-77c-41.7-13.6-72-52.8-72-99s30.3-85.5 72-99V264c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v221c41.7 13.6 72 52.8 72 99s-30.3 85.5-72 99zm279.6-143.9c.2 0 .3-.1.4-.1v221c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V539c.2 0 .3.1.4.1-42-13.4-72.4-52.7-72.4-99.1 0-46.4 30.4-85.7 72.4-99.1-.2 0-.3.1-.4.1v-77c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v77c-.2 0-.3-.1-.4-.1 42 13.4 72.4 52.7 72.4 99.1 0 46.4-30.4 85.7-72.4 99.1zM616 440a36 36 0 1072 0 36 36 0 10-72 0zM403.4 566.5l-1.5-2.4c0-.1-.1-.1-.1-.2l-.9-1.2c-.1-.1-.2-.2-.2-.3-1-1.3-2-2.5-3.2-3.6l-.2-.2c-.4-.4-.8-.8-1.2-1.1-.8-.8-1.7-1.5-2.6-2.1h-.1l-1.2-.9c-.1-.1-.3-.2-.4-.3-1.2-.8-2.5-1.6-3.9-2.2-.2-.1-.5-.2-.7-.4-.4-.2-.7-.3-1.1-.5-.3-.1-.7-.3-1-.4-.5-.2-1-.4-1.5-.5-.4-.1-.9-.3-1.3-.4l-.9-.3-1.4-.3c-.2-.1-.5-.1-.7-.2-.7-.1-1.4-.3-2.1-.4-.2 0-.4 0-.6-.1-.6-.1-1.1-.1-1.7-.2-.2 0-.4 0-.7-.1-.8 0-1.5-.1-2.3-.1s-1.5 0-2.3.1c-.2 0-.4 0-.7.1-.6 0-1.2.1-1.7.2-.2 0-.4 0-.6.1-.7.1-1.4.2-2.1.4-.2.1-.5.1-.7.2l-1.4.3-.9.3c-.4.1-.9.3-1.3.4-.5.2-1 .4-1.5.5-.3.1-.7.3-1 .4-.4.2-.7.3-1.1.5-.2.1-.5.2-.7.4-1.3.7-2.6 1.4-3.9 2.2-.1.1-.3.2-.4.3l-1.2.9h-.1c-.9.7-1.8 1.4-2.6 2.1-.4.4-.8.7-1.2 1.1l-.2.2a54.8 54.8 0 00-3.2 3.6c-.1.1-.2.2-.2.3l-.9 1.2c0 .1-.1.1-.1.2l-1.5 2.4c-.1.2-.2.3-.3.5-2.7 5.1-4.3 10.9-4.3 17s1.6 12 4.3 17c.1.2.2.3.3.5l1.5 2.4c0 .1.1.1.1.2l.9 1.2c.1.1.2.2.2.3 1 1.3 2 2.5 3.2 3.6l.2.2c.4.4.8.8 1.2 1.1.8.8 1.7 1.5 2.6 2.1h.1l1.2.9c.1.1.3.2.4.3 1.2.8 2.5 1.6 3.9 2.2.2.1.5.2.7.4.4.2.7.3 1.1.5.3.1.7.3 1 .4.5.2 1 .4 1.5.5.4.1.9.3 1.3.4l.9.3 1.4.3c.2.1.5.1.7.2.7.1 1.4.3 2.1.4.2 0 .4 0 .6.1.6.1 1.1.1 1.7.2.2 0 .4 0 .7.1.8 0 1.5.1 2.3.1s1.5 0 2.3-.1c.2 0 .4 0 .7-.1.6 0 1.2-.1 1.7-.2.2 0 .4 0 .6-.1.7-.1 1.4-.2 2.1-.4.2-.1.5-.1.7-.2l1.4-.3.9-.3c.4-.1.9-.3 1.3-.4.5-.2 1-.4 1.5-.5.3-.1.7-.3 1-.4.4-.2.7-.3 1.1-.5.2-.1.5-.2.7-.4 1.3-.7 2.6-1.4 3.9-2.2.1-.1.3-.2.4-.3l1.2-.9h.1c.9-.7 1.8-1.4 2.6-2.1.4-.4.8-.7 1.2-1.1l.2-.2c1.1-1.1 2.2-2.4 3.2-3.6.1-.1.2-.2.2-.3l.9-1.2c0-.1.1-.1.1-.2l1.5-2.4c.1-.2.2-.3.3-.5 2.7-5.1 4.3-10.9 4.3-17s-1.6-12-4.3-17c-.1-.2-.2-.4-.3-.5z"
            />
          </svg>
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-yp8pcc ant-btn-primary ant-btn-dangerous filter-btn"
        type="button"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
            />
          </svg>
        </span>
      </button>
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-yp8pcc"
      >
        <div
          class="ant-row css-dev-only-do-not-override-yp8pcc"
        >
          <div
            class="ant-col ant-col-sm-24 ant-col-md-8 css-dev-only-do-not-override-yp8pcc"
          >
            <div
              class="ant-card ant-card-bordered product-item css-dev-only-do-not-override-yp8pcc"
            >
              <div
                class="ant-card-cover"
              >
                <a
                  href="/123"
                >
                  <img
                    src="bluetooth-speaker.jpg"
                  />
                </a>
              </div>
              <div
                class="ant-card-body"
              >
                <div
                  class="ant-card-meta"
                >
                  <div
                    class="ant-card-meta-detail"
                  >
                    <div
                      class="ant-card-meta-title"
                    >
                      <div
                        class="product-title"
                      >
                        Bluetooth Speaker
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-card-meta"
                >
                  <div
                    class="ant-card-meta-detail"
                  >
                    <div
                      class="ant-card-meta-title"
                    >
                      $123.45
                    </div>
                  </div>
                </div>
              </div>
              <ul
                class="ant-card-actions"
              >
                <li
                  style="width: 50%;"
                >
                  <span>
                    <span
                      class="product-number"
                    >
                      #
                      1
                    </span>
                  </span>
                </li>
                <li
                  style="width: 50%;"
                >
                  <span>
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                      tabindex="-1"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="ant-col ant-col-sm-24 ant-col-md-8 css-dev-only-do-not-override-yp8pcc"
          >
            <div
              class="ant-card ant-card-bordered product-item css-dev-only-do-not-override-yp8pcc"
            >
              <div
                class="ant-card-cover"
              >
                <a
                  href="/234"
                >
                  <img
                    src="gamepad.jpg"
                  />
                </a>
              </div>
              <div
                class="ant-card-body"
              >
                <div
                  class="ant-card-meta"
                >
                  <div
                    class="ant-card-meta-detail"
                  >
                    <div
                      class="ant-card-meta-title"
                    >
                      <div
                        class="product-title"
                      >
                        Gamepad
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-card-meta"
                >
                  <div
                    class="ant-card-meta-detail"
                  >
                    <div
                      class="ant-card-meta-title"
                    >
                      $234.56
                    </div>
                  </div>
                </div>
              </div>
              <ul
                class="ant-card-actions"
              >
                <li
                  style="width: 50%;"
                >
                  <span>
                    <span
                      class="product-number"
                    >
                      #
                      2
                    </span>
                  </span>
                </li>
                <li
                  style="width: 50%;"
                >
                  <span>
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                      tabindex="-1"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
